beetsplug/acousticbrainz.py

```diff
@@ -42,127 +42,127 @@ def commands(self):
 
         def func(lib, opts, args):
             items = lib.items(ui.decargs(args))
-            fetch_info(self._log, items, ui.should_write())
+            self._fetch_info(items, ui.should_write())
 
         cmd.func = func
         return [cmd]
 
     def import_task_files(self, session, task):
         """Function is called upon beet import.
         """
+        self._fetch_info(task.imported_items(), False)
 
-        items = task.imported_items()
-        fetch_info(self._log, items, False)
-
-
-def fetch_info(log, items, write):
-    """Get data from AcousticBrainz for the items.
-    """
-
-    def get_value(*map_path):
-        try:
-            return reduce(operator.getitem, map_path, data)
-        except KeyError:
-            log.debug(u'Invalid Path: {}', map_path)
+    def _fetch_info(self, items, write):
+        """Get data from AcousticBrainz for the items.
+        """
 
-    for item in items:
-        if item.mb_trackid:
-            log.info(u'getting data for: {}', item)
+        def get_value(*map_path):
+            try:
+                return reduce(operator.getitem, map_path, data)
+            except KeyError:
+                log.debug(u'Invalid Path: {}', map_path)
+
+        for item in (item for item in items if item.mb_trackid):
+            self._log.info(u'getting data for: {}', item)
+            data = self._get_data(item.mb_trackid)
+            if data:
+                # Get each field and assign it on the item.
+                item.danceable = get_value(
```

Comment:
I'm sorry I tried my best to convince git this whole patch did not change as much as git thinks it did, but to no avail.